README.md

```diff
@@ -300,3 +300,27 @@ All commits should pass the pre-commit hooks. Enable them to run before each
 commit:
 
     ln -s ../../hooks/pre-commit .git/hooks/pre-commit
+
+Running with Docker
+======================
+Getting all the dependencies right to build and run logcabin can be an issue. Its easier to run it with a docker container where dependency versions are managed properly. There is a dockerfile which can be used as following
```

Comment:
Some small formatting nitpicks. I'm not sure if they are enforced but it couldn't hurt to match the rest of the file.

1) Looks like all the section headers have a new line after them.
2) Looks like a column length limit is respected except for URLs (probaly 80).

Overall I was able to build in the docker container specified here.

I did get 1 unit test failure when running tests in the container.

```
root@acd8a7262ee0:/logcabin# ./build/test/test 
[==========] Running 423 tests from 58 test cases.
1595531506.861324 RPC/Address.cc:213 in refresh() WARNING[12:thread 1]: Unknown error from getaddrinfo("qqq", "81"): Temporary failure in name resolution
[----------] 10 tests from RPCClientRPCTest
[ RUN      ] RPCClientRPCTest.waitForReply_timeout
build/RPC/ClientRPCTest.cc:159: Failure
Value of: rpc.waitForReply(__null, __null, ClientRPC::Clock::now() + std::chrono::seconds(10))
  Actual: TIMEOUT
Expected: ClientRPC::Status::OK
Which is: OK
[==========] 423 tests from 58 test cases ran. (15939 ms total)
[  PASSED  ] 422 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] RPCClientRPCTest.waitForReply_timeout

 1 FAILED TEST
```

Reply:
Fixed the column width. 
I tried to reproduce this error but could not. I tried two things.
1. Limiting resources to docker container to see if it fails some tests.

docker run --rm -it --memory 128m --cpuset-cpus 0 --cpus=0.5 -v $(pwd):/logcabin logcabin-dev:latest /bin/bash -c /logcabin/build/test/test

This test depends on timing, so failures are likely under heavy load, valgrind, etc.
[==========] 601 tests from 67 test cases ran. (5724 ms total)
[  PASSED  ] 600 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] ClientClientImplExactlyOnceTest.keepAliveThreadMain_TimingSensitive

2. Trying turning off the internet. This fails RPCAddressTest
uild/RPC/AddressTest.cc:184: Failure
Value of: googleDNS
  Actual: "Unspecified"
Expected: "8.8.8.8:80"
This test requires connectivity to the Internet for a DNS lookup. Alternatively, you can point google-public-dns-a.google.com to 8.8.8.8 in your /etc/hosts file.
[==========] 601 tests from 67 test cases ran. (173608 ms total)
[  PASSED  ] 600 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] RPCAddressTest.refresh